connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoSourceOptions.scala

```diff
@@ -23,4 +24,19 @@ object KustoSourceOptions extends KustoOptions {
   val KUSTO_BLOB_STORAGE_SAS_URL: String = newOption("blobStorageSasUrl")
   // Blob container name
   val KUSTO_BLOB_CONTAINER: String = newOption("blobContainer")
+
+  // By default an estimation of the rows count is first being made, if the count is lower than 5000 records a simple
+  // query is made, else - if storage params were provided they are used for scale reading and if not - the connector
+  // tries to use storage from the kusto ingest service.
+  // This option allows to override these connector heuristics.
+  // In default mode if a lean mode was chosen and failed - there is a fallback to scale mode
+  // See https://docs.microsoft.com/en-us/azure/kusto/concepts/querylimits#limit-on-result-set-size-result-truncation
+  // for hard limit on query size using lean mode
+  val KUSTO_READ_MODE: String = newOption("readMode")
 }
+
+object ReadMode extends Enumeration {
+  type ReadMode = Value
+  val Default, ForceLeanMode, ForceScaleMode = Value
```

Comment:
Default is not a good value for an Enum, because I have no way of understanding what default is in the context of `ReadMode` ...

Reply:
I dont think i can pass the logic by the name here - the huristics is described in the options ... i will add it to the docs

Reply:
how about - `BestEffort`, or `Optimized`? (I don't like the term Default because it is a reserved keyword in a lot of languages)

Reply:
i really prefer the default actually - also discussed this with boaz .. your suggestions suggest that there is a third option . i really prefer it being "Default", "Regular" or maybe just drop it and instead check if the option was provided ..

Reply:
Ok, so check if it was provided. 
Default is not a real enum value.